clippy_lints/src/methods/mod.rs

```diff
@@ -2588,6 +2617,39 @@ fn lint_iter_next<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx hir::Expr<'_>, iter_
     }
 }
 
+fn lint_iter_count<'tcx>(cx: &LateContext<'tcx>, expr: &Expr<'_>, iter_args: &'tcx [Expr<'tcx>], is_mut: bool) {
```

Comment:
Could you move the function to its own file (maybe `iter_count.rs`) as `bytes_nth` does in order not to make `methods/mod.rs` messy anymore? Please see #6680 for details.

Reply:
I've just moved the function to `iter_count.rs`!